### PR TITLE
Closes #4021: Use save image candidate be used with video and audio elements

### LINKED ISSUE #4021: Let "Save Image" canidate be used with video and audio elements
Originally filed as mozilla-mobile/fenix#3703.

Users expect to be able to download video and audio files too. The save image canidate should be updated to work with those types.

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
Waiting for string approval.

**Comment by rocketsroger** [comment]
device-2020-02-26-125001

**Comment by rocketsroger** [comment]
@AmyYLee Please approve context menu behaviour/string.  Thanks,

**Comment by rocketsroger** [comment]
> @AmyYLee Please approve context menu behaviour/string. Thanks,

I opened an issue to track the "about:blank" title.  https://github.com/mozilla-mobile/android-components/issues/6090

File: components/feature/contextmenu/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -17,6 +17,8 @@
     <string name="mozac_feature_contextmenu_copy_image_location">Copy image location</string>
     <!-- Text for context menu item to save / download the image. -->
     <string name="mozac_feature_contextmenu_save_image">Save image</string>
+    <!-- Text for context menu item to save / download the video / audio. -->
+    <string name="mozac_feature_contextmenu_save_video_audio">Save video/audio</string>
```
Comment (pocmo) [inline components/feature/contextmenu/src/main/res/values/strings.xml]: Doesn't it make sense to have two strings (or candidates) here? Seems weird to use "video/audio" if we know what it is. :)

Comment (rocketsroger) [reply components/feature/contextmenu/src/main/res/values/strings.xml]: I agree.  But from testing in some cases audio is returning `HitResult.VIDEO` so it says `Save video` when its supposed to say `Save audio`.  Looks like when the audio is being played the site might use the video player.

**Comment by AmyYLee** [comment]
@betsymi for string review/approval here.

@rocketsroger Can you give me more context of how this screen is triggered?

**Comment by rocketsroger** [comment]
> @rocketsroger Can you give me more context of how this screen is triggered?

Sure.  This is triggered when the user long tap on the video or audio player. (while the media is already playing)

**Comment by rocketsroger** [comment]
device-2020-02-27-113202
What this looks like on Fenix

**Comment by AmyYLee** [comment]
> > @rocketsroger Can you give me more context of how this screen is triggered?
> 
> Sure. This is triggered when the user long tap on the video or audio player. (while the media is already playing)

@betsymi - Some context here for you after talk to @rocketsroger is that we can't consistently tell if it's an audio file that is being downloaded. They can always tell if it's a video file though. There's been some situations where it will show "Save video" when it's actually audio so right now they have it labeled as "Save video/audio". 

I was thinking maybe we say "Save file" which is more generic if we can't 100% tell. Let me know what you think would be the best string solution here. Thanks!

**Comment by rocketsroger** [comment]
device-2020-02-27-172201

I've found an solution to correctly display the link.  I've also updated the string as Amy's suggestion.  Please let me know what you think.

**Comment by betsymi** [comment]
@AmyYLee 'Save file' is a good solution. I also had a thought that users may wonder where it's being saved. Would `Save file to device` help clarify do you think? I know it's a tad longer.

**Comment by AmyYLee** [comment]
@betsymi yeah I think "Save file the device" helps make it clearer.

**Comment by rocketsroger** [comment]
@betsymi @AmyYLee Thanks! I will update the string and merge this change.

**Comment by rocketsroger** [comment]
device-2020-02-28-112753

**Comment by rocketsroger** [comment]
bors r=Amejia481

**Pull request merged**